Framework/Algorithms/src/CropWorkspaceRagged.cpp

```diff
@@ -0,0 +1,142 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier: GPL - 3.0 +
+#include "MantidAlgorithms/CropWorkspaceRagged.h"
+#include "MantidAPI/MatrixWorkspace.h"
+#include "MantidAPI/WorkspaceFactory.h"
+#include "MantidKernel/ArrayProperty.h"
+#include "MantidKernel/BoundedValidator.h"
+#include "MantidKernel/MandatoryValidator.h"
+
+namespace {
+std::vector<double> getSubVector(Mantid::MantidVec &data,
+                                 const int64_t &lowerIndex,
+                                 const int64_t &upperIndex) {
+  auto low = std::next(data.begin(), lowerIndex);
+  auto up = std::next(data.begin(), upperIndex);
+  // get new vectors
+  std::vector<double> newData(low, up);
+  return newData;
+}
+
+} // namespace
+
+namespace Mantid {
+namespace Algorithms {
+
+using namespace Kernel;
+using namespace API;
+
+DECLARE_ALGORITHM(CropWorkspaceRagged)
+
+/// Init function
+void CropWorkspaceRagged::init() {
+  declareProperty(std::make_unique<WorkspaceProperty<MatrixWorkspace>>(
+                      "InputWorkspace", "", Direction::Input),
+                  "The input workspace");
+  declareProperty(std::make_unique<WorkspaceProperty<MatrixWorkspace>>(
+                      "OutputWorkspace", "", Direction::Output),
+                  "Name to be given to the cropped workspace.");
+
+  auto required = std::make_shared<MandatoryValidator<std::vector<double>>>();
+  declareProperty(std::make_unique<ArrayProperty<double>>("XMin", required),
+                  ".");
```

Comment:
Could we have a better description for the docs instead of just .

Reply:
good spot, I missed that